### PR TITLE
feat!: move table to query param & rename existing headers and query params

### LINKED ISSUE #1771: Shape API tweaks
A few things have come up:

## Proposal: move table from its current place as the last path segment to just another (required) query argument.

The root table for a shape is currently in the API path e.g. `v1/shape/${table_name}`. This is a required parameter and my original logic to do it this way was basically that it's easy to write when testing out simple shape per table shapes. E.g. `curl http://localhost:3000/v1/shape/foo`.

But it does get a bit weird as people might need to specify the schema the table is part of. Also there's an odd chance that in the future, we'll support alternative backends that don't have "tables". Also we might want to have some sort of multi-table syntax. Regardless, moving the table out of the path gives us more flexibility.

So that foo shape curl above would look like:

`curl http://localhost:3000/v1/shape?table=foo`

## Proposal: change `shape_id` to `shape_handle`

The problem with `id` is it strongly suggests that it's a permanent identifier for the particular shape. Whereas it's just pointing to a particular ephemeral instantiation of the shape. A particular shape instantiation be removed at any moment so we need a term that conveys the idea of a unique identifier that you can use to access or manipulate a specific instance of the shape log.

`handle` is often used for this sort of thing.

`token` is another term I considered but it has too much whiff of session-y or authorization related usage.

### PR DESCRIPTION
**Pull request by thruflo**
Included renames:
- `electric-chunk-last-offset` -> `electric-offset`
- `electric-next-cursor` -> `electric-cursor`
- `electric-shape-id` -> `electric-handle`
- `?shape_id` -> `?handle`

Added query parameter: `?table` instead of `/:table` in the path

Closes #1771, #1796, #1798

### CODE REVIEW

File: integration-tests/tests/crash-recovery.lux [hunk 1]
```diff
@@ -41,8 +41,8 @@
 [shell client]
   # strip ANSI codes from response for easier matching
   !curl -v -X GET http://localhost:3000/v1/shape/items?offset=-1
-  ?electric-shape-id: ([\d-]+)
-  [local shape_id=$1]
+  ?electric-shape-handle: ([\d-]+)
```
Comment (msfstef) [inline integration-tests/tests/crash-recovery.lux:44]:
```suggestion
  ?electric-handle: ([\d-]+)
```

Comment (thruflo) [reply integration-tests/tests/crash-recovery.lux:44]: Yeah, what is that `.lux` file? I left it alone in case it was some kind of artefact.

Comment (msfstef) [reply integration-tests/tests/crash-recovery.lux:44]: We have a set of integration tests using lux - this one tests crash recovery

**Comment by thruflo** [comment]
Btw, I don't understand why the typecheck tests are failing. `pnpm typecheck` runs fine for me locally in both the `react-hooks` and `typescript-client` packages:

```
➜  react-hooks git:(thruflo/header-naming) ✗ pnpm run typecheck

> @electric-sql/react@0.4.7 typecheck /Users/thruflo/Development/electric-sql/electric/packages/react-hooks
> tsc -p tsconfig.json

➜  react-hooks git:(thruflo/header-naming) ✗ cd ..
➜  packages git:(thruflo/header-naming) ✗ cd typescript-client
➜  typescript-client git:(thruflo/header-naming) ✗ pnpm run typecheck

> @electric-sql/client@0.6.5 typecheck /Users/thruflo/Development/electric-sql/electric/packages/typescript-client
> tsc -p tsconfig.json
```

**Comment by msfstef** [comment]
@thruflo the `react-hooks` package depends on the typescript client, so you need to first `pnpm buiild` the typescript client and then run `pnpm typecheck` on the react hooks package

**Comment by thruflo** [comment]
Yup, I've `pnpm -r build`ed from the repo root locally.

I'm not really the right person to revise the way we run our CI tests :)

**Comment by msfstef** [comment]
@thruflo I was also seeing the same issue - typecheck passed locally but failed on CI, and the lines referenced on CI for the errors did not match the actual test file, so something odd happening there...

There's a conflict now so tests aren't running though.

**Comment by thruflo** [comment]
@msfstef fixed the conflict.

**Comment by msfstef** [comment]
@thruflo types fixed 💪 🚀

**Review by icehaunter — APPROVED** [review]

**Pull request merged**